waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -337,9 +337,12 @@
   (testing-using-waiter-url
     (when (using-k8s? waiter-url)
       (let [current-user (retrieve-username)]
-        (testing "Has service account with matching default namespace"
+        (testing "No service account for default namespace, or matches user"
           (let [service-account (get-pod-service-account waiter-url nil)]
-            (is (= current-user service-account))))
+            ;; blank when default namespace is set to a specific user
+            ;; matches run-as-user when default-namespace=*
+            (is (or (str/blank? service-account)
```

Comment:
Should we have an if-else checking default namespace is `*` and then have separate assertions?

Reply:
Changed to choose the assertion based on pod's `k8s/namespace` property.